### PR TITLE
Validate onchain operator ID against local operator ID

### PR DESCRIPTION
**Pull request by ian-shim**
## Why are these changes needed?
Suspend node if the registered operator ID onchain is different from the operator ID locally loaded from BLS key.

## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -216,6 +221,22 @@ func (n *Node) Start(ctx context.Context) error {
...
+	if operator != nil && operator.Address != "" {
+		operatorID, err := n.Transactor.OperatorAddressToID(ctx, gethcommon.HexToAddress(operator.Address))
+		if err != nil {
+			return fmt.Errorf("failed to get operator ID: %w", err)
+		}
+		if operatorID != operator.OperatorId {
+			return fmt.Errorf("operator ID mismatch: expected %s, got %s", operator.OperatorId.Hex(), operatorID.Hex())
+		}
+	} else {
+		// Doesn't have operator address. Just check that operator ID is registered onchain
```
Comment (jianoaix) [inline node/node.go:233]: This part may need to be dropped, we are moving towards a model where operators first bring up the node (without registration)

Comment (ian-shim) [reply node/node.go:233]: What happens when operators first bring up the node without registration? It just sits and does nothing?

Comment (mooselumph) [reply node/node.go:233]: Yes. They can check aspects of their setup (reachability, etc), without immediately being accountable for data. This should help address problem of operators joining and being down for a few hours.

Comment (ian-shim) [reply node/node.go:233]: Got it. Makes sense

**Review by jianoaix — APPROVED** [review]

**Pull request merged**